iml-agent/src/daemon_plugins/journal.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{
+    agent_error::ImlAgentError,
+    daemon_plugins::{DaemonPlugin, Output},
+};
+use futures::{lock::Mutex, Future, FutureExt, Stream, TryStreamExt};
+use iml_wire_types::{JournalMessage, JournalPriority};
+use std::{io, pin::Pin, str, sync::Arc};
+use tokio::io::stream_reader;
+use tokio_util::codec::{FramedRead, LinesCodec};
+
+#[derive(Debug, PartialEq, serde::Deserialize)]
+#[serde(untagged)]
+enum Msg {
+    String(String),
+    Bytes(Vec<u8>),
+}
+
+#[derive(Debug, PartialEq, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct IncomingMessage {
+    #[serde(rename = "__CURSOR")]
+    pub cursor: String,
+    #[serde(rename = "__REALTIME_TIMESTAMP")]
+    pub realtime_timestamp: String,
+    #[serde(rename = "PRIORITY")]
+    pub priority: Option<JournalPriority>,
+    #[serde(rename = "SYSLOG_FACILITY")]
+    pub syslog_facility: Option<String>,
+    #[serde(rename = "SYSLOG_IDENTIFIER")]
+    pub syslog_identifier: Option<String>,
+    #[serde(rename = "MESSAGE")]
+    pub message: Msg,
+}
+
+#[derive(Debug)]
+pub struct Journal {
+    cursor: Arc<Mutex<Option<String>>>,
+}
+
+pub fn create() -> impl DaemonPlugin {
+    Journal {
+        cursor: Arc::new(Mutex::new(None)),
+    }
+}
+
+async fn read_entries(
+    cursor: &Option<String>,
+) -> Result<impl Stream<Item = Result<(String, JournalMessage), ImlAgentError>>, ImlAgentError> {
+    let client = reqwest::Client::new();
+
+    let range = if let Some(x) = cursor {
+        format!("entries={}:1:1000", x)
+    } else {
+        "entries=:-1000:".into()
+    };
+
+    let s = client
+        .get("http://localhost:19531/entries")
```

Comment:
Maybe it's worth a command line option for the port at least.

Reply:
Sure. I'll add an environment var for it.